server/src/test/java/org/opensearch/extensions/ExtensionsOrchestratorTests.java

```diff
@@ -477,54 +485,11 @@ public void testHandleActionListenerOnFailureRequest() throws Exception {
-        Path extensionDir = createTempDir();
-        Files.write(extensionDir.resolve("extensions.yml"), extensionsYmlLines, StandardCharsets.UTF_8);
-        ExtensionsOrchestrator extensionsOrchestrator = new ExtensionsOrchestrator(settings, extensionDir);
-        initialize(extensionsOrchestrator);
-
-        List<Setting<?>> componentSettings = List.of(
-            Setting.boolSetting("falseSetting", false, Property.IndexScope, Property.NodeScope),
-            Setting.simpleString("fooSetting", "foo", Property.Dynamic)
-        );
-
-        // Test EnvironmentSettingsRequest arg constructor
-        EnvironmentSettingsRequest environmentSettingsRequest = new EnvironmentSettingsRequest(componentSettings);
-        List<Setting<?>> requestComponentSettings = environmentSettingsRequest.getComponentSettings();
-        assertEquals(componentSettings.size(), requestComponentSettings.size());
-        assertTrue(requestComponentSettings.containsAll(componentSettings));
-        assertTrue(componentSettings.containsAll(requestComponentSettings));
-
-        // Test EnvironmentSettingsRequest StreamInput constructor
-        try (BytesStreamOutput out = new BytesStreamOutput()) {
-            environmentSettingsRequest.writeTo(out);
-            out.flush();
-            try (BytesStreamInput in = new BytesStreamInput(BytesReference.toBytes(out.bytes()))) {
-                environmentSettingsRequest = new EnvironmentSettingsRequest(in);
-                requestComponentSettings = environmentSettingsRequest.getComponentSettings();
-                assertEquals(componentSettings.size(), requestComponentSettings.size());
-                assertTrue(requestComponentSettings.containsAll(componentSettings));
-                assertTrue(componentSettings.containsAll(requestComponentSettings));
-            }
-        }
-
-    }
-
     public void testEnvironmentSettingsResponse() throws Exception {
 
-        List<Setting<?>> componentSettings = List.of(
-            Setting.boolSetting("falseSetting", false, Property.IndexScope, Property.NodeScope),
-            Setting.simpleString("fooSetting", "foo", Property.Dynamic)
-        );
-
         // Test EnvironmentSettingsResponse arg constructor
-        EnvironmentSettingsResponse environmentSettingsResponse = new EnvironmentSettingsResponse(settings, componentSettings);
-        assertEquals(componentSettings.size(), environmentSettingsResponse.getComponentSettingValues().size());
-
-        List<Setting<?>> responseSettings = new ArrayList<>();
-        responseSettings.addAll(environmentSettingsResponse.getComponentSettingValues().keySet());
-        assertTrue(responseSettings.containsAll(componentSettings));
-        assertTrue(componentSettings.containsAll(responseSettings));
+        EnvironmentSettingsResponse environmentSettingsResponse = new EnvironmentSettingsResponse(settings);
+        assertEquals(settings, environmentSettingsResponse.getEnvironmentSettings());
```

Comment:
We should add a test to compare if the setting present in the list matches the one we have added to the list

Reply:
With this modification to Environment settings support, we are no longer transporting a list of `Setting<T>` objects but rather an entire `Settings` object, so we are not adding anything to any list. As for the comparison test for this transported `Settings` object, this is handled on line 492.